ci/Jenkinsfile

```diff
@@ -0,0 +1,84 @@
+import groovy.transform.Field
+
+def defVal (value, defaultValue) {
+  if (value == null || value == []) {
+    return defaultValue
+  } else {
+    if (value instanceof java.util.LinkedHashMap) {
+      return value + defaultValue
+    } else {
+      return value
+    }
+  }
+}
+
+@Field def defaultEnv = ["CI=true"]
+@Field def defaultTest = "go test -v ./..."
+
+def call(opts = []) {
+  def env = defVal(opts['env'], defaultEnv)
+  def test = defVal(opts['test'], defaultTest)
+  stage('tests') {
+    parallel(
+      linux: {
+        node(label: 'linux') {
+          ansiColor('xterm') {
+            def root = tool name: '1.9.2', type: 'go'
```

Comment:
We are currently running 1.10, do we have to duplicate this?

Reply:
It was mostly copied from https://github.com/ipfs/jenkins-libs/blob/master/vars/golang.groovy, the jenkinsfile here in POC which I don't want merged (at least definitely not in this state)

Reply:
Yes, is absolutely POC and super messy atm. Look at the javascript one for a preview of final state: https://github.com/ipfs/jenkins-libs/blob/master/vars/javascript.groovy, notice the top variables defining platforms + versions to test with.